File: python/520.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if len(word) > 1 and word[0].islower() and word[1].isupper():
            return False
        return all(word[i].isupper() == word[1].isupper() for i in range(2, len(word)))

File: go/520.go
package main

import "unicode"

func detectCapitalUse(word string) bool {
	cnt := 0
	for _, ch := range word {
		if unicode.IsUpper(ch) {
			cnt++
		}
	}
	return cnt == 0 || cnt == len(word) || cnt == 1 && unicode.IsUpper(rune(word[0]))
}
